inflate.c

```diff
@@ -1157,9 +1158,13 @@ int ZEXPORT inflate(z_stream *strm, int flush) {
                 copy = left;
             left -= copy;
             state->length -= copy;
-            do {
-                *put++ = *from++;
-            } while (--copy);
+            if (copy >= sizeof(long)) {
+                put = chunk_copy(put, from, copy);
```

Comment:
I think the easy way to fix the comment in memcopy.h is to duplicate this code, and put different versions in the if and else blocks around line 1153 above.  Since the first case can never come from the output buffer, it can never be a `memset()` operation, so it's simpler.